src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpointImpl.java

```diff
@@ -0,0 +1,242 @@
+package de.hpi.bpt.argos.api.entityType;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import de.hpi.bpt.argos.api.eventType.EventTypeEndpointImpl;
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.attribute.type.TypeAttribute;
+import de.hpi.bpt.argos.storage.dataModel.entity.type.EntityType;
+import de.hpi.bpt.argos.storage.dataModel.mapping.EventEntityMapping;
+import de.hpi.bpt.argos.storage.dataModel.mapping.MappingCondition;
+import de.hpi.bpt.argos.util.HttpStatusCodes;
+import de.hpi.bpt.argos.util.RestEndpointUtil;
+import de.hpi.bpt.argos.util.RestEndpointUtilImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static spark.Spark.halt;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EntityTypeEndpointImpl implements EntityTypeEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final Gson serializer = new Gson();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.get(EntityTypeEndpoint.getEntityTypeHierarchyBaseUri(), this::getEntityTypeHierarchy);
+        sparkService.get(EntityTypeEndpoint.getEntityTypeAttributesBaseUri(), this::getEntityTypeAttributes);
+        sparkService.get(EntityTypeEndpoint.getEntityTypeEntityMappingsBaseUri(), this::getEntityTypeEntityMappings);
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEntityTypeHierarchy(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        List<EntityType> allEntityTypes = PersistenceAdapterImpl.getInstance().getEntityTypes();
+        if (allEntityTypes == null) {
+            halt(HttpStatusCodes.ERROR, "could not read entity types.");
+        }
+        EntityType root = null;
+        Map<Long, List<EntityType>> tree = new HashMap<>();
+
+        // Convert list to tree
+        for (EntityType entityType : allEntityTypes) {
+            if (entityType.getParentId() == -1) {

```

Comment:
This is actually not sufficent. There might be multiple `EventTypes` with `ParentId == -1`.